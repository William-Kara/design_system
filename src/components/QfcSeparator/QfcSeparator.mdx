import { Meta } from '@storybook/blocks';

<Meta title="Elements/Separator" />

# Definition

QfcSeparator component is typically used to separate sections or between elements.

# Usage

To use QfcSeparator, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcSeparator.scss';

```
