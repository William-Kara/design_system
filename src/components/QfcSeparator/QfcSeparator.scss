.qfc-separator {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  background-color: $border;
  border: none;
}