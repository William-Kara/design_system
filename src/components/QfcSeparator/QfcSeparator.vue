<template>
  <hr
    class="qfc-separator"
    :style="{
      height: thickness,
    }"
  />
</template>

<script lang="ts" setup>
interface Props {
  thickness: string;
}
withDefaults(defineProps<Props>(), {
  thickness: '1px',
});
</script>

<style lang="scss" scoped src="./QfcSeparator.scss" />
