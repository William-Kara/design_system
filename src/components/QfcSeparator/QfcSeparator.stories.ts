import QfcSeparator from './QfcSeparator.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof QfcSeparator> = {
  component: QfcSeparator,
  title: 'Elements/Separator',
  argTypes: {
    thickness: {
      control: {
        type: 'text',
      },
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {},
};

export const Icon: Story = {
  args: {
    thickness: '2px',
  },
};
