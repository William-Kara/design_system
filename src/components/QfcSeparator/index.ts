import QfcSeparator from './QfcSeparator.vue';

export { QfcSeparator };
export default QfcSeparator;
