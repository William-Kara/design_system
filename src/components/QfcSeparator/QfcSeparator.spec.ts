import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcSeparator from './QfcSeparator.vue';

let wrapper: any = null;

const propsValues = {
  thickness: '2px',
};

beforeEach(() => {
  wrapper = shallowMount(QfcSeparator, {
    propsData: propsValues,
  });
});

describe('QfcSeparator Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
