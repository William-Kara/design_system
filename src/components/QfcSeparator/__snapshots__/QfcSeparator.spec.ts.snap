// Vitest Snapshot v1

exports[`QfcSeparator Component > should render correctly 1`] = `"<hr data-v-32b8b850=\\"\\" class=\\"qfc-separator\\" style=\\"height: 2px;\\">"`;
